packages/purpose-process/src/model/domain/apiConverter.ts

```diff
@@ -0,0 +1,114 @@
+import { match } from "ts-pattern";
+import {
+  Purpose,
+  PurposeRiskAnalysisForm,
+  PurposeVersion,
+  PurposeVersionDocument,
+  PurposeVersionState,
+  RiskAnalysisMultiAnswer,
+  RiskAnalysisSingleAnswer,
+  purposeVersionState,
+} from "pagopa-interop-models";
+import {
+  ApiPurpose,
+  ApiPurposeVersion,
+  ApiPurposeVersionDocument,
+  ApiPurposeVersionState,
+  ApiRiskAnalysisForm,
+} from "./models.js";
+
+export const singleAnswersToApiSingleAnswers = (
+  singleAnswers: RiskAnalysisSingleAnswer[]
+): Record<string, string[]> =>
+  singleAnswers.reduce<Record<string, string[]>>((acc, curr) => {
+    if (!curr.value) {
+      return acc;
+    }
+    // eslint-disable-next-line functional/immutable-data
+    acc[curr.key] = [curr.value];
+    return acc;
+  }, {});
+
+export const multiAnswersToApiMultiAnswers = (
+  multiAnswers: RiskAnalysisMultiAnswer[]
+): Record<string, string[]> =>
+  multiAnswers.reduce<Record<string, string[]>>((acc, curr) => {
+    // eslint-disable-next-line functional/immutable-data
```

Comment:
```suggestion
    if (!curr.values) {
      return acc;
    }
    // eslint-disable-next-line functional/immutable-data
```
You can add this check here too

Reply:
Note: in the case above it was required because `curr.value` could be `undefined`. Here `curr.values` is an array: maybe we should check if it's empty and skip it. Anyway `!curr.values` will not work as you wish because it would always be true because the array is never `undefined` (but maybe it's empty, so should we check that case?)

Reply:
> Note: in the case above it was required because `curr.value` could be `undefined`. Here `curr.values` is an array: maybe we should check if it's empty and skip it. Anyway `!curr.values` will not work as you wish because it would always be true because the array is never `undefined` (but maybe it's empty, so should we check that case?).

You're right! I think it would be enough to just check it's length!

Reply:
Updated